### PR TITLE
Porting TokenizingTextBox

### PR DESCRIPTION
**Pull request by niels9001**
This PR ports the `TokenizingTextBox` to the new repo, addressing: #53

Changes:
- Ported over code, sample and tests.
- Visual update to the `TokenizingTextBox` to make it inline with WinUI styling:

Before:
image

After:
image

- Working on WASDK and UWP - couldn't figure out why it's not showing the tokens on WASM. Might be the `ItemsPresenter` issue we were having with `SettingsExpander` as well?

### CODE REVIEW

**Comment by niels9001** [comment]
@Arlodotexe When copying the `WrapPanel` code manually into the project, everything works. When referencing the `.Primitives.proj` I'm getting this error: 

image

**Review by Arlodotexe — APPROVED** [review]

**Comment by niels9001** [comment]
@Arlodotexe did you see the crash as mentioned above?

**Comment by Arlodotexe** [comment]
> @Arlodotexe did you see the crash as mentioned above?

It looks like you are referencing the primitives .csproj directly via the MSBuild property `ToolkitPrimitiveSourceProject`, but I'm not seeing the crash you mentioned on any platform. 

Uno does appear to have an issue though. We should file a ticket to investigate this issue so it can be fixed from Uno's end.

**Review by michael-hawker — COMMENTED** [review]
Didn't try testing locally yet, but just going through a few things to get up to speed. Did a comparison with Uno fork and didn't notice any big differences. I know originally, I had tested this control on Android for the Graph components, but don't think I had tried WASM at the time.

File: components/TokenizingTextBox/src/CommunityToolkit.WinUI.Controls.TokenizingTextBox.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <Version>0.0.1</Version>
+
+    <!-- Rns suffix is required for namespaces shared across projects. See https://github.com/CommunityToolkit/Labs-Windows/issues/152 -->
+    <RootNamespace>CommunityToolkit.WinUI.Controls.TokenizingTextBoxRns</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="$(ToolkitExtensionSourceProject)" />
+    <ProjectReference Include="$(ToolkitHelperSourceProject)" />
+    <ProjectReference Include="$(ToolkitPrimitiveSourceProject)" />
```
[the remark this answers is not in the document]
Comment (Arlodotexe) [reply components/TokenizingTextBox/src/CommunityToolkit.WinUI.Controls.TokenizingTextBox.csproj:14]: This was one of the things we were unsure about, glad to clear that up 👍

Think I'll open a separate PR and clean them all up.

File: components/TokenizingTextBox/samples/TokenizingTextBox.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+## Events
+
+| Events | Description |
+| -- | -- |
+| QuerySubmitted | Event raised when the user submits the text query. |
+| SuggestionChosen | Event raised when a suggested item is chosen by the user. |
+| TextChanged | Event raised when the text input value has changed. |
+| TokenItemAdding | Event raised before a new token item has been added. Can be used to transform user text into an object. |
+| TokenItemRemoving | Event raised before a token item is removed (cancelable). |
+| TokenItemRemoved | Event raised after a token item has been removed. |
```
[the remark this answers is not in the document]
Comment (Arlodotexe) [reply components/TokenizingTextBox/samples/TokenizingTextBox.md]: Cleaned up in fdfcb9090d6753cb2089c3dbe58b77d2a418a3b7 so we can merge now and do docs later.

File: components/TokenizingTextBox/src/TokenizingTextBox.Selection.cs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+    }
+
+    internal void SelectAllTokensAndText()
+    {
+#if WINAPPSDK
+            _ = DispatcherQueue.EnqueueAsync(
+#else
+            var dispatcherQueue = DispatcherQueue.GetForCurrentThread();	
+            _ = dispatcherQueue.EnqueueAsync(
+#endif
```
[the remark this answers is not in the document]
Comment (Arlodotexe) [reply components/TokenizingTextBox/src/TokenizingTextBox.Selection.cs]: Fixed in f60895c3eb40163f524512f224449f72bff241be

**Pull request merged**